python/lsst/ts/ledprojector/led_controller.py

```diff
@@ -0,0 +1,495 @@
+        # Note that we use enumerate to get the numerical index
+        a = [
+            LabjackChannel(serialNumber=data[0], channel=data[1], index=ind)
+            for ind, data in enumerate(tmp, start=0)
+        ]
+        # Now create our multi-key dictionary
+        # TODO: write unit test for this dictionary setup
+        self.channels: Dict[Union[str, int], LabjackChannel] = {}
+        for lbc in a:
+            self.channels.update({lbc.serial: lbc})
+            self.channels.update({lbc.index: lbc})
+            self.channels.update({lbc.channel: lbc})
+
+        self.log.info(f"led_controller dictionary {self.channels}")
+
+        # Set if connected to the labjack and state data seen,
+        # cleared otherwise.
+        self.status_event = asyncio.Event()
+        self.status_task = utils.make_done_future()
+
+    @classmethod
+    def get_config_schema(cls) -> dict[str, Any]:
+        return yaml.safe_load(
+            """
+$schema: http://json-schema.org/draft-07/schema#
+description: Schema for LEDController
+type: object
+properties:
+  device_type:
+    description: LabJack model
+    type: string
+    default: T7
+  connection_type:
+    description: Connection type
+    type: string
+    default: TCP
+  identifier:
+    description: >-
+        LabJack indentifier:
+        * A host name or IP address if connection_type = TCP or WIFI
+        * A serial number if connection_type = USB
+        * For testing in an environment with only one LabJack you may use ANY
+    type: string
+  poll_interval:
+    description: Polling interval (seconds)
+    type: number
+    default: 1
+  topics:
+    description: >-
+      Array of batches of relevant sensors.
+    type: array
+    minItems: 1
+    items:
+      types: object
+      minItems: 1
+      properties:
+        topic_name:
+            description: Casual name for the sensor cluster.
+            type: string
+        sensor_name:
+            description: Value for the sensor_name field of the topic.
+            type: string
+        location:
+            description: >-
+                Location of sensors. A comma-separated list, with one item per non-null channel_name.
+            type: string
+        led_names:
+            description: >-
+                Names of LEDs for identification, such as the serial number.
+            type: array
+            minItems: 1
+            items:
+              type: string
+        channel_names:
+            description: >-
+                LabJack channel names, in order of the field array.
+            type: array
+            minItems: 1
+            items:
+              type: string
+      required:
+        - topic_name
+        - sensor_name
+        - location
+        - led_names
+        - channel_names
+required:
+  - device_type
+  - connection_type
+  - identifier
+  - poll_interval
+additionalProperties: false
+"""
+        )
+
+    async def run(self) -> None:
```

Comment:
Are you sure you need to overwrite the parent class method? It seems like it is basically doing the same thing (with the addition of the "TODO's"

Reply:
I agree, although I get this error when running pytest:
`TypeError: Can't instantiate abstract class LEDController with abstract method run`

Reply:
Oh right! I was looking at `LabJackDataClient` and not `BaseLabJackDataClient`, which is the class you are subclassing.. So I think you are good!